grammar.js

```diff
@@ -215,7 +215,28 @@ module.exports = grammar({
       )
     )),
 
+    _literal: $ => choice(
+      $.symbol,
+      $.integer,
+      $.float,
+      $.boolean,
+      $.nil,
+      $.string
+    ),
+
     symbol: $ => token(seq(':', choice(identifierChars(), operatorChars()))),
+    integer: $ => (/0b[01](_?[01])*|0[oO]?[0-7](_?[0-7])*|(0d)?\d(_?\d)*|0x[0-9a-fA-F](_?[0-9a-fA-F])*/),
+    float: $ => (/\d(_?\d)*\.\d(_?\d)*([eE]\d(_?\d)*)?/),
+    boolean: $ => choice('true', 'false', 'TRUE', 'FALSE'),
+    nil: $ => choice('nil', 'NIL'),
+
+    string: $ => choice(
+      $._single_quoted,
+      $._double_quoted
+    ),
+    _single_quoted: $ => (/'(\\.|[^\\'])*'/),
+    _double_quoted: $ => token(seq('"', repeat(choice(/\\./, $.interpolation, /[^\\"]/)), '"')),
+    interpolation: $ => seq('#{', $._expression, '}'),
```

Comment:
@maxbrunsfeld: With `token`, the `#{…}` interpolation is parsed, but we don’t get any nodes generated for the `interpolation` & `_expression` productions. Without `token`, the `#` gets interpreted as the beginning of a comment, which breaks parsing.

It looks like the comment rule gets applied before trying to parse the rules, except within tokens; thus, the `/\\./` regexp is able to correctly parse a backslash-escaped `#{…}` as being part of a flat string.

Would it be reasonable to instead attempt to parse the comment/whitespace rules when matching the current rule fails?